pallets/builder-incentives/src/types.rs

```diff
@@ -0,0 +1,99 @@
+use frame_support::{
+	pallet_prelude::*,
+	traits::{fungible::Inspect, Currency},
+};
+use pallet_transaction_payment::OnChargeTransaction;
+use sp_core::H160;
+use sp_runtime::traits::{Saturating, Zero};
+
+use crate::Config;
+
+pub type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
+/// AccountContractId for pallet_revive
+pub type AccountContractId = H160;
+/// Era number type
+pub type EraNumber = u32;
+/// An index to a block.
+pub type BlockNumber = u32;
+
+/// Type aliases used for interaction with `OnChargeTransaction`.
+pub(crate) type OnChargeTransactionOf<T> =
+	<T as pallet_transaction_payment::Config>::OnChargeTransaction;
+
+/// OnChargeTransactionBalance type alias.
+pub(crate) type OnChargeTransactionBalanceOf<T> =
+	<OnChargeTransactionOf<T> as OnChargeTransaction<T>>::Balance;
+
+pub(crate) type ContractsOf<T> = <T as pallet_revive::Config>::Currency;
```

Comment:
Weird type alias - contracts referring to currency associated type.

Reply:
You are right, no need for it.